src/main/java/dev/pgm/community/sessions/services/SQLSessionService.java

```diff
@@ -0,0 +1,202 @@
+package dev.pgm.community.sessions.services;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import dev.pgm.community.database.DatabaseConnection;
+import dev.pgm.community.feature.SQLFeatureBase;
+import dev.pgm.community.sessions.Session;
+import dev.pgm.community.sessions.SessionQuery;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import tc.oc.pgm.util.concurrent.ThreadSafeConnection.Query;
+
+public class SQLSessionService extends SQLFeatureBase<Session, SessionQuery> {
+
+  private static final String TABLE_NAME = "sessions";
+  private static final String TABLE_FIELDS =
+      "(id VARCHAR(36) PRIMARY KEY, player VARCHAR(36), disguised BOOL, server VARCHAR(32), startTime LONG, endTime LONG)";
+
+  private LoadingCache<SessionQuery, SelectLatestSessionQuery> sessionCache;
+
+  public SQLSessionService(DatabaseConnection database) {
+    super(database, TABLE_NAME, TABLE_FIELDS);
+    this.sessionCache =
+        CacheBuilder.newBuilder()
+            .build(
+                new CacheLoader<SessionQuery, SelectLatestSessionQuery>() {
+                  @Override
+                  public SelectLatestSessionQuery load(SessionQuery key) throws Exception {
```

Comment:
```suggestion
                  public SelectLatestSessionQuery load(@Nonnull SessionQuery key) {
```
the throws is redundant, and the API says it will be nonnull, should annotate it too